plugin/evm/customheader/dynamic_fee_state.go

```diff
@@ -43,7 +44,17 @@ func feeStateBeforeBlock(
 		}
 	}
 
-	state.AdvanceTime(timestamp - parent.Time)
+	switch {
+	case config.IsGranite(timestamp):
+		parentMS := parent.Time * 1000
+		if config.IsGranite(parent.Time) {
+			parentMS = *customtypes.GetHeaderExtra(parent).TimeMilliseconds
+		}
```

Comment:
This logic appears repeated in a few places..is there a clean/easy way to move it into a `header.GetTimeMS()` type of helper?